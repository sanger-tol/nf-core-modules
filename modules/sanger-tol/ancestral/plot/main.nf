process ANCESTRAL_PLOT {
    label 'process_low'
    tag "$meta.id"

    // Docker image available at the project github repository
    container "ghcr.io/sanger-tol/busco_painter:1.0.1"

    input:
    tuple val(meta), path(comp_location)
    tuple val(meta2), path(fai)

    output:
    path("*_buscopainter.png")  , emit: png_plot
    path("*_buscopainter.pdf")  , emit: pdf_plot
    path("versions.yml")        , emit: versions

    script:
    if (workflow.profile.tokenize(',').intersect(['conda', 'mamba']).size() >= 1) {
        error "ANCESTRAL_PLOT module does not support Conda. Please use Docker / Singularity / Podman instead."
    }

    def args    = task.ext.args     ?: ''
    def prefix  = task.ext.prefix   ?: "${meta.id}"

    """
    plot_buscopainter.R \\
        -f ${comp_location} \\
        -p ${prefix} \\
        -i ${fai} \\
        ${args}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        R: \$(R --version | grep -oP "\\d+\\.\\d+\\.\\d+")
        plot_buscopainter.R: \$(plot_buscopainter.R -v)
    END_VERSIONS
    """

    stub:
    def args    = task.ext.args     ?: ''
    def prefix  = task.ext.prefix   ?: "${meta.id}"
    """
    touch ${comp_location}_buscopainter.png
    touch ${comp_location}_buscopainter.pdf

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        R: \$(R --version | grep -oP "\\d+\\.\\d+\\.\\d+")
        plot_buscopainter.R: \$(plot_buscopainter.R -v)
    END_VERSIONS
    """
}
