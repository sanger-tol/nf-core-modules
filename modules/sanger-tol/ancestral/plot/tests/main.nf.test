
nextflow_process {

    name "Test Process ANCESTRAL_PLOT"
    script "../main.nf"
    process "ANCESTRAL_PLOT"

    tag "modules"
    tag "modules_sangertol"
    tag "ancestral"
    tag "ancestral/plot"

    config './nextflow.config'

    test("GCA_963859965.1 - tsv - fai") {

        when {
            params {
                outdir  = "$outputDir"
            }

            process {
                """
                input[0] = Channel.of(
                    [
                        [ id:'test' ], // meta map
                        file(params.modules_testdata_base_path + 'resources/genomenote/ancestral_data/buscopainter_complete_location.tsv', checkIfExists: true)
                    ]
                )

                input[1] = Channel.of(
                    [
                        [ id:'test' ], // meta map
                        file(params.modules_testdata_base_path + 'resources/genomenote/ancestral_data/GCA_963859965.1.fasta.fai', checkIfExists: true)
                    ]
                )
                """
            }
        }

        then {
            def stable_name_files = getAllFilesFromDir(
                params.outdir,
                includeDir: true,
                relative: true,
                include: [
                    // Meta Data changes between runs
                    "*.pdf",
                    "*.png"
                ]
            )

            assertAll(
                { assert process.success },
                { assert snapshot(
                    stable_name_files,
                    process.out.versions
                ).match() }
            )
        }

    }

    test("GCA_963859965.1 - tsv - fai - stub") {

        options "-stub"

        when {
            params {
                outdir  = "$outputDir"
            }

            process {
                """
                input[0] = Channel.of(
                    [
                        [ id:'test' ], // meta map
                        file(params.modules_testdata_base_path + 'resources/genomenote/ancestral_data/buscopainter_complete_location.tsv', checkIfExists: true)
                    ]
                )

                input[1] = Channel.of(
                    [
                        [ id:'test' ], // meta map
                        file(params.modules_testdata_base_path + 'resources/genomenote/ancestral_data/GCA_963859965.1.fasta.fai', checkIfExists: true)
                    ]
                )
                """
            }
        }

        then {
            def stable_name_files = getAllFilesFromDir(
                params.outdir,
                includeDir: true,
                relative: true,
                include: [
                    // Meta Data changes between runs
                    "*.pdf",
                    "*.png"
                ]
            )

            assertAll(
                { assert process.success },
                { assert snapshot(
                    stable_name_files,
                    process.out.versions
                ).match() }
            )
        }

    }

}
