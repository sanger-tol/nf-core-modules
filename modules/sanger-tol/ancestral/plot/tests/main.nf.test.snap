{
    "GCA_963859965.1 - tsv - fai - stub": {
        "content": [
            [
                
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-10-06T11:33:52.784498374"
    },
    "GCA_963859965.1 - tsv - fai": {
        "content": [
            [
                
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-10-06T11:33:44.603114364"
    }
}